### PR TITLE
Adds countries and state filter options for queries

### PR DESCRIPTION
**Pull request by ka7eh**

### CODE REVIEW

**Comment by brendan-ward** [comment]
@ka7eh thanks for doing this.

Could you please change the API a bit, to make it more extensible?

Instead of `countries` and `admin` as top-level properties, what about adding this as an object instead:

`geonamesParams: {
    country: ['us', 'ca'],
    adminCode1: ['or', 'wa', 'bc']
}`

This allows the user to easily add `adminCode2` for instance.

I also prefer the codes as separate strings rather than comma delimited.

You can add [this](http://www.geonames.org/export/geonames-search.html) as a cross-reference for the supported API fields.

File: L.Control.Geonames.js [hunk 1]
```diff
@@ -134,12 +140,12 @@ L.Control.Geonames = L.Control.extend({
             li.innerText = 'No results found';
         }
     },
-    _getName: function(geoname){
+    _getName: function (geoname) {
```
Comment (brendan-ward) [inline L.Control.Geonames.js:143]: This isn't working right for multiple codes.  It is currently sending them through as a comma-delimited list, but per the API, these need to have one query parameter for each code:
`...&adminCode1=OR&adminCode1=WV`

File: example.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            this._input.type = 'text';
+            this._input.placeholder = 'Enter a state';
+            return this._container;
+        }
+    });
+    map.addControl(new stateField());
+
+    var control = L.control.geonames({
+        username: 'your-geonames-username',
+        countries: 'us,ca',
```
Comment (brendan-ward) [inline example.html:41]: Can you add this as an text entry too?  This way results will be more obvious and makes it easier to merge with `index.html`.  (meaning: if user does not enter country or adminCode1, then results are equivalent to current version).

**Comment by brendan-ward** [comment]
Also note: the `gh-pages` branch has an index.html page that demonstrates this, but not `master`.  Maybe we should merge your example.html and that one into a single `index.html` on master (I can take care of merging it into `gh-pages`).

**Comment by ka7eh** [comment]
See: https://github.com/consbio/Leaflet.Geonames/pull/5

**Pull request closed without merging**